### PR TITLE
Fix ChainNotify RPC

### PR DESCRIPTION
**Pull request by ec2**
**Summary of changes**
Changes introduced in this pull request:
- Fixes Chain Notify subscription IDs, storage miner no longer throwing warnings. 

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: blockchain/chain/src/store/chain_store.rs [hunk 1]
```diff
@@ -992,7 +993,9 @@ pub async fn sub_head_changes(
                 .await;
         }
     });
-    let cancel_sender = events_pubsub.write().await.subscribe();
+    let mut x = events_pubsub.write().await;
+    let cancel_sender = x.subscribe();
+    drop(x);
```
Comment (austinabell) [inline blockchain/chain/src/store/chain_store.rs:998]: Can't you just do this in one line? You don't need to keep the lock for anything?

File: node/rpc/src/chain_api.rs [hunk 2]
```diff
@@ -69,10 +69,10 @@ where
     KS: KeyStore + Send + Sync + 'static,
     B: Beacon + Send + Sync + 'static,
 {
-    let (data_subscribe, _) = data.state_manager.chain_store().subscribe().await;
+    let (data_subscribe, ts) = data.state_manager.chain_store().subscribe().await;
     let index = chain::sub_head_changes(
         data_subscribe,
-        &data.state_manager.chain_store().heaviest_tipset().await,
+        &Some(ts),
```
Comment (austinabell) [inline node/rpc/src/chain_api.rs:75]: why is this an option if this is the only place it's used? Should subscribe just return the option then?

**Review by austinabell — APPROVED** [review]

File: blockchain/chain/src/store/chain_store.rs (hunk 1 above)
Comment (ec2) [reply blockchain/chain/src/store/chain_store.rs:998]: Ah yeah, i wasnt sure at one point if the lock was being held by calling subscribe, but yeah I can remove this.

File: blockchain/chain/src/store/chain_store.rs [hunk 3]
```diff
@@ -967,20 +967,17 @@ where
 
 pub async fn sub_head_changes(
     mut subscribed_head_change: Subscriber<HeadChange>,
-    heaviest_tipset: &Option<Arc<Tipset>>,
+    heaviest_tipset: &Arc<Tipset>,
```
Comment (austinabell) [inline blockchain/chain/src/store/chain_store.rs:970]: also, does it need to be a reference if it can just be moved? it's cloned during the subscribe?

**Pull request merged**